client/client-multi/client-c/test/kaa_tcp_channel/test_kaa_tcp_channel_bootstrap.c

```diff
@@ -730,16 +730,23 @@ kaa_error_t kaa_platform_protocol_alloc_serialize_client_sync(kaa_platform_proto
 
 ext_tcp_socket_state_t ext_tcp_utils_tcp_socket_check(kaa_fd_t fd, const kaa_sockaddr_t *destination, kaa_socklen_t destination_size)
 {
+    int result_memcmp;
```

Comment:
The only rule that affects variable scope in MISRA-2012 is `Rule 8.9 An object should be defined at block scope if its identifier only appears in a single function`.